middleware-jackson/src/main/java/com/nytimes/android/external/store3/middleware/jackson/JacksonBufferedSourceAdapter.kt

```diff
@@ -0,0 +1,16 @@
+package com.nytimes.android.external.store3.middleware.jackson
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.nytimes.android.external.fs3.BufferedSourceAdapter
+import okio.BufferedSource
+import okio.Okio
+import java.io.ByteArrayInputStream
+import javax.inject.Inject
+
+/**
+ * An implementation of [BufferedSourceAdapter] that uses [ObjectMapper] to convert Java values to JSON.
+ */
+class JacksonBufferedSourceAdapter<Parsed> @Inject
```

Comment:
The Java class was also containing a logic that would propagate a thrown exception! Is that deleted in purpose on the Kotlin conversion?

Reply:
Yes! @tiwiz mentioned failing fast, not sure 100% if that is the desired direction. @digitalbuddha

Reply:
that would be a breaking change imo, leave the same as before pls